evm_loader/cli/src/main.rs

```diff
@@ -66,6 +122,453 @@ fn emulate(solana_url: String, evm_loader: Pubkey, contract_id: H160, caller_id:
+                    last_valid_slot.saturating_sub(slot)
+                ));
+            }
+
+            if pending_transactions.is_empty() {
+                return Ok(());
+            }
+
+            if slot > last_valid_slot {
+                break;
+            }
+
+            let epoch_info = rpc_client.get_epoch_info()?;
+            tpu_addresses = get_leader_tpus(
+                min(epoch_info.slot_index + 1, epoch_info.slots_in_epoch),
+                NUM_TPU_LEADERS,
+                leader_schedule.as_ref(),
+                cluster_nodes.as_ref(),
+            );
+
+            for transaction in pending_transactions.values() {
+                if !tpu_addresses.is_empty() {
+                    let wire_transaction =
+                        serialize(&transaction).expect("serialization should succeed");
+                    for tpu_address in &tpu_addresses {
+                        send_transaction_tpu(&send_socket, &tpu_address, &wire_transaction);
+                    }
+                } else {
+                    let _result = rpc_client
+                        .send_transaction_with_config(
+                            transaction,
+                            RpcSendTransactionConfig {
+                                preflight_commitment: Some(commitment.commitment),
+                                ..RpcSendTransactionConfig::default()
+                            },
+                        )
+                        .ok();
+                }
+            }
+
+            if cfg!(not(test)) {
+                // Retry twice a second
+                sleep(Duration::from_millis(500));
+            }
+        }
+
+        if send_retries == 0 {
+            return Err("Transactions failed".into());
+        }
+        send_retries -= 1;
+
+        // Re-sign any failed transactions with a new blockhash and retry
+        let (blockhash, _fee_calculator, new_last_valid_slot) = rpc_client
+            .get_recent_blockhash_with_commitment(commitment)?
+            .value;
+        last_valid_slot = new_last_valid_slot;
+        transactions = vec![];
+        for (_, mut transaction) in pending_transactions.into_iter() {
+            transaction.try_sign(signer_keys, blockhash)?;
+            transactions.push(transaction);
+        }
+    }
+}
+
+fn command_deploy(
+    config: &Config,
+    program_location: &str,
+) -> CommandResult {
+
+    let ACCOUNT_HEADER_SIZE = 1+Account::SIZE;
+    let CONTRACT_HEADER_SIZE = 1+Contract::SIZE;
+
+    let program_data = read_program_data(program_location)?;
+    let program_code_len = CONTRACT_HEADER_SIZE + program_data.len() + 2*1024;
+    let minimum_balance_program = config.rpc_client.get_minimum_balance_for_rent_exemption(ACCOUNT_HEADER_SIZE)?;
+    let minimum_balance_code = config.rpc_client.get_minimum_balance_for_rent_exemption(program_code_len)?;
+    let minimum_caller_balance = config.rpc_client.get_minimum_balance_for_rent_exemption(ACCOUNT_HEADER_SIZE)?;
```

Comment:
minimum_balance_program and minimum_caller_balance will have the same value. Remove the duplicated RPC request.

Beside this, minimum_caller_balance is unused variable and can be completely removed.

Reply:
You are right. Fixed.